stf/src/sgx.rs

```diff
@@ -259,16 +269,45 @@ impl Stf {
                     )
                         .encode(),
                 ));
+                increment_nonce(&account_incognito);
                 Ok(())
             }
             TrustedCall::balance_shield(who, value) => {
                 debug!("balance_shield({:x?}, {})", who.encode(), value);
-                Self::shield_funds(who, value)?;
+                if !validate_nonce(&who, call.nonce) {
+                    return Err(StfError::InvalidNonce(call.nonce))
+                }
+                Self::shield_funds(who.clone(), value)?;
+                increment_nonce(&who);
                 Ok(())
             }
+        }
+    )
+    }
+
+    pub fn account_nonce(ext: &mut State, account: &AccountId) -> Option<Vec<u8>> {
+        ext.execute_with(|| {
+                if let Some(info) = get_account_info(account) {
+                    debug!("Account {:?} nonce is {}",account.encode(), info.nonce);
+                    Some(info.nonce.encode())
+            } else {
+                None
+            }
         })
     }
 
+    pub fn account_data(ext: &mut State, account: &AccountId) -> Option<Vec<u8>> {
```

Comment:
this function is never used. please remove

Reply:
It is used for unit testing

Reply:
What I could do is merging account_data and account_nonce to a single function returning the account information. Or add a test feature, as proposed in issue #215

Reply:
the test feature would make it clear. But just providing account info is fine too